### PR TITLE
(chore) add `name` attribute to all language grammars

### LINKED ISSUE #2394: [chore] Moves names of languages into data of grammar itself
Before, from the header:

```js
/*
Language: Java
Author: Vsevolod Solovyov <vsevolod.solovyov@gmail.com>
...
*/
```

**Note, we're not changing the headers, just copying the data into the grammar definitions.**

After (in the grammar definition itself):

```js
return {
    name: 'Java',
    aliases: ['jsp'],
    keywords: KEYWORDS,
    illegal: /<\/|#/,
```

I think it makes sense for name to be the very first key.  It rarely changes and might make resolving merge conflicts easier sometimes.

### PR DESCRIPTION
**Pull request by taufik-nurrohman**
Closes #2394 

Related: #2400

### CODE REVIEW

File: src/languages/c-like.js [hunk 1]
```diff
@@ -197,6 +197,7 @@ export default function(hljs) {
   };
 
   return {
+    name: 'C-like',
```
Comment (yyyc514) [inline src/languages/c-like.js]: This is just a foundation. No name needed.

File: src/languages/apache.js [hunk 2]
```diff
@@ -19,7 +19,8 @@ export default function(hljs) {
     begin: ":\\d{1,5}"
   };
   return {
-    aliases: ['apacheconf'],
+    name: 'Apache',
```
Comment (yyyc514) [inline src/languages/apache.js]: Please add “config” to match the title.

File: src/languages/julia.js [hunk 3]
```diff
@@ -86,6 +86,7 @@ export default function(hljs) {
 
   // placeholder for recursive self-reference
   var DEFAULT = {
+    name: 'Julia',
```
Comment (yyyc514) [inline src/languages/julia.js]: This needs to go in the return block not default.

File: src/languages/pf.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
-Language: packet filter config
```
Comment (yyyc514) [inline src/languages/pf.js]: Is packet filter an actual proper noun here?

File: src/languages/yaml.js [hunk 5]
```diff
@@ -45,8 +45,9 @@ export default function(hljs) {
   };
 
   return {
+    name: 'YAML',
     case_insensitive: true,
-    aliases: ['yml', 'YAML', 'yaml'],
```
Comment (yyyc514) [inline src/languages/yaml.js]: Can we add this back?  It’s not really part of the name change and it’s not consistent either. We need to look into how we handle case with aliases and then approach this problem across the board.

**Comment by yyyc514** [comment]
Very nice work. Thanks for the contribution.

**Comment by yyyc514** [comment]
You really should rebase this on master though.  I meant I can work with this, but for future contributions the PR will be MUCH nicer and easier to review if you're starting from a clean master.

File: src/languages/pf.js (hunk 4 above)
Comment (taufik-nurrohman) [reply src/languages/pf.js]: Looks proper to be written as PF alone maybe?

<https://en.wikipedia.org/wiki/PF_(firewall)>

Comment (yyyc514) [reply src/languages/pf.js]: From that article it looks like your change here is fine.  Resolving this.

**Comment by taufik-nurrohman** [comment]
> Also please remove the extra changelog file you seemed to add.

Done? Huh? Where am I now?

**Comment by taufik-nurrohman** [comment]
> You really should rebase this on master though. I meant I can work with this, but for future contributions the PR will be MUCH nicer and easier to review if you're starting from a clean master.

My first intention was that I wanted to merge this commit onto your branch in  [`yyyc514:lang_has_name`](https://github.com/yyyc514/highlight.js/tree/lang_has_name). But I just didn&rsquo;t know how it worked.

**Comment by yyyc514** [comment]
lang_has_name is in master now and this commit chain is just crazy.

You should hard reset your own master to upstream master and then cherry-pick your 2 commits back on top. (that's easiest if you don't know how to do the rebase)

**Comment by yyyc514** [comment]
> Wow, this project is so crowded. I feel lost sometimes.

Sorry Git and Github can be a real pain if you aren't familiar with them.

**Comment by taufik-nurrohman** [comment]
> Have you been merging upstream master into your master (while also committing to your master)?

That was from [my previous pull request](https://github.com/highlightjs/highlight.js/pull/2372). I didn&rsquo;t delete that repository after my pull has been merged and just continue creating a branch named `patch-1`.

The commit log data in my branch is mixed with yours. Is that normal?

**Comment by yyyc514** [comment]
> That was from my previous pull request. I didn’t delete that repository after my pull has been merged and just continue creating a branch named patch-1.

Yeah that's how you get everything messed up.  Because the final merge of anything often isn't the same as what's in a PR... so now your history has diverged from the main branch and you start seeing everything twice... to avoid that your master should always be pinned to upstream master.

You can fix it by a hard reset:

```sh
# make sure upstream is a remote
git remote add upstream git@github.com:highlightjs/highlight.js.git
# checkout your master branch
git checkout master 
# force reset your master to point to the SAME exact commit as upstream/master
git reset --hard upstream/master
```

That's just to "fix" master... you should always do your work in branches and NEVER commit to master (when contributing to projects).  If you keep your master clean then all you have to do is switch to it periodically and merge upstream:

```
git checkout master
git merge upstream/master
```

Once you have your master sane again you can switch to your branch, hard-reset IT to master also and then `cherry-pick` the two commits you want to get them back.  Make sure you write down the commit hashes first (or you can always use `git reflog` to find them).

**Comment by yyyc514** [comment]
If that's' too much let me know and I can probably finish this up myself.

**Comment by taufik-nurrohman** [comment]
Doh, sorry for that `extensions` part. I should move that to another pull. Please wait.

**Pull request merged**